src/XrmMockupShared/Requests/QualifyLeadRequestHandler.cs

```diff
@@ -0,0 +1,153 @@
+                }
+            }
+
+            EntityMetadata customerMetadata = null;
+            if (request.OpportunityCustomerId != null)
+            {
+                customerMetadata = metadata.EntityMetadata.GetMetadata(request.OpportunityCustomerId.LogicalName);
+                if (customerMetadata == null)
+                {
+                    throw new FaultException($"The entity with a name = '{request.OpportunityCustomerId.LogicalName}' was not found in the MetadataCache.");
+                }
+            }
+
+            var lead = db.GetDbRow(request.LeadId);
+
+            if (lead == null || request.LeadId.LogicalName != "lead")
+            {
+                throw new FaultException($"lead With Id = {request.LeadId.Id} Does Not Exist");
+            }
+
+            var currentState = lead.GetColumn<int>("statecode");
+
+            if (currentState == 1)
+            {
+                throw new FaultException("The lead is already closed.");
+            }
+
+            var status = request.Status != null ? request.Status.Value : 0;
+            var statusOptionMetadata = Utility.GetStatusOptionMetadata(leadMetadata)
+                .FirstOrDefault(s => s.Value == status) as StatusOptionMetadata;
+
+            if (statusOptionMetadata == null)
+            {
+                throw new FaultException($"{status} is not a valid status code on lead with Id {request.LeadId.Id}");
+            }
+
+            if (statusOptionMetadata.State != 1)
+            {
+                throw new FaultException($"{request.Status.Value} is not a valid status code for state code LeadState.Qualified lead with Id {request.LeadId.Id}");
+            }
+
+            var createdEntities = new EntityReferenceCollection();
+
+            if (request.CreateAccount)
+            {
+                var accountAlreadyExists = db["account"].Any(row => row["name"] != null && row["name"].Equals(lead["fullname"]));
+
+                if (accountAlreadyExists)
+                {
+                    throw new FaultException("A record was not created or updated because a duplicate of the current record already exists.");
+                }
+
+                var account = new Entity
+                {
+                    Id = Guid.NewGuid(),
+                    LogicalName = "account",
+                };
+
+                createdEntities.Add(account.ToEntityReference());
+            }
+
+            if (request.CreateContact)
+            {
+                var contactAlreadyExists = db["contact"].Any(row => row["fullname"] != null && row["fullname"].Equals(lead["fullname"]));
+
+                if (contactAlreadyExists)
+                {
+                    throw new FaultException("A record was not created or updated because a duplicate of the current record already exists.");
+                }
+
+                var contact = new Entity
```

Comment:
Qualify Lead uses fields on the lead when creating the contact such as Contact Name and Job title.